src/_guides/language/sound-dart.md

```diff
@@ -0,0 +1,696 @@
+
+<div class="fails-sa" markdown="1">
+{% prettify dart %}
+var x = 3;    // x is inferred as an int
+x = 4.0;
+{% endprettify %}
+</div>
+
+<div class="passes-sa" markdown="1">
+{% prettify dart %}
+num y = 3; // y is defined as num which can be double or int
+y = 4.0;
+{% endprettify %}
+</div>
+
+### Type argument inference
+
+Type arguments to constructor calls and generic method invocations are
+inferred based on a combination of downward information from the context
+of occurrence, and upwards information from the arguments to the constructor
+or generic method. If inference is not doing what you want or expect,
+you can always explicitly pass the type arguments.
+
+<div class="passes-sa" markdown="1">
+{% prettify dart %}
+// Inferred as if you wrote <int>[].
+List<int> listOfInt = [];
+
+// Inferred as if you wrote <double>[3.0].
+var listOfDouble = [3.0];
+
+// x is inferred as int, double using downward information.
+// Return type of the closure is inferred as int using upwards information.
+// Type argument to map() is inferred as <int> using upwards information.
+var listOfInt2 = listOfDouble.map((x) => x.toInt());
+{% endprettify %}
+</div>
+
+## How to enable strong mode
+
+Dart's static analysis engine enforces type safety. You can enable
+strong mode using one of the following approaches:
+
+* Use an analysis options file
+* Call dartanalyzer with the strong mode flag
+* Enable strong mode in DartPad
+
+### Use an analysis options file
+
+Create an analysis options file at the package root of your project.
+Besides turning on strong mode, you can also enable any of the
+available linter rules. For more information, see
+[Customize Static Analysis](/guides/language/analysis-options).
+
+### Call dartanalyzer with strong mode enabled
+
+The [dartanalyzer](https://github.com/dart-lang/sdk/tree/master/pkg/analyzer_cli#dartanalyzer)
+tool supports several flags related to strong mode:
+
+| --[no-]strong | Enable strong static checks |
+| --no-implicit-casts | Disable implicit casts in strong mode |
+| --no-implicit-dynamic | Disable implicit dynamic |
+{:.table .table-striped .nowrap}
+
+For more information on these flags, see
+[Specifying strong mode](/guides/language/analysis-options#specifying-strong-mode).
+
+<aside class="alert alert-info" markdown="1">
+**Note:** An analysis options file in the same directory where you call
+dartanalyzer overrides the command line flag. So, if the analysis options
+file disables strong mode, you can't enable it using `--strong`.
+</aside>
+
+### Enable strong mode in DartPad
+
+If you use [DartPad](/tools/dartpad) to write and test code, you can
+enable strong mode by checking the **Strong mode** box in the lower
+right corner. Note that DartPad doesn't support the implicit casts flag,
+implicit dynamic flag, or enabling linter rules. For this functionality
+you must use DDC or IntelliJ for now.
+
+## Substituting types
+
+When can you replace a type with a subtype or a supertype?
```

Comment:
I'll have to object to the entire paragraph. It's mixing different usages of the word "substitution", and in an OO setting you should only ever use one meaning: Lisskov's Substitution Principle, which says that you can always use an instance of a subclass where an instance of a superclass is expected. (More pedantically, it says that any visible property of instances of the supertype must also hold for instances of the subtype).

This is substitution at the *value* level - a value of one type can be used instead of a value of another type. We represent that in the language by making subtype values assignable to supertype'd variables. We *allow* that assignment because the principle tells us that it's safe.

When you "override", you are not replacing anything. You *create* a subclass which has a different type for that method. You want/need the subclass to be a subtype, therefore you are required to make the new method type substitutable for the supertype method's type.

Passing a parameter to a function is not substitution at all (unless we are doing this as lambda calculus). It's about assignability - you assign a value to a variable.

Now, it's late and I haven't read the document yet, so I'm not sure what the section is supposed to say, so I don't have brilliant suggestion for what to write instead. I just know that this paragraph got me completely confused about what it is trying to say, probably because "substituting types" is meaningless to me :)